src/nucleus/jdg.ml

```diff
@@ -804,11 +801,10 @@ let occurs_eq_term_abstraction = occurs_abstraction TT.assumptions_eq_term
 
 let apply_abstraction inst_u sgn abstr e0 =
   match abstr with
-  | TT.NotAbstract _ ->
-     failwith "Tried to apply an argument to a NotAbstract. Type-checking should prevent this!"
+  | TT.NotAbstract _ -> TT.error TT.AbstractionExpected
```

Comment:
This should be a failwith / assert false because AML type-checking keeps count of abstraction levels.